### PR TITLE
adding the ability for the user to enter the custom count down timer.

### PR DESCRIPTION
**Pull request by parchea**

### CODE REVIEW

File: SmartWeights/SmartWeights/Workout/StartWorkout.swift [hunk 1]
```diff
@@ -215,6 +226,17 @@ class WorkoutViewModel: ObservableObject {
                             if self.WorkoutState == .final{
                                 self.finishWorkout()
                                 print("Workout stopped. workoutInProgress: \(self.workoutInProgress)")
+                                // Cancel the recognition task before stopping the audio engine
```
Comment (AdamRa97) [inline SmartWeights/SmartWeights/Workout/StartWorkout.swift:229]: How come Line 229 - Line 239 is commented out? Is this on purpose?

File: SmartWeights/SmartWeights/Workout/StartWorkout.swift [hunk 2]
```diff
@@ -384,6 +397,18 @@ class WorkoutViewModel: ObservableObject {
             isWorkingOut = false
         }
         
+        // Logic for completing the workout
```
Comment (AdamRa97) [inline SmartWeights/SmartWeights/Workout/StartWorkout.swift:400]: Should these be commented out as well?

File: SmartWeights/SmartWeights/Workout/StartWorkout.swift [hunk 3]
```diff
@@ -438,9 +463,10 @@ class WorkoutViewModel: ObservableObject {
     
     func playSound() {
         do {
-
-            try AVAudioSession.sharedInstance().setCategory(.playAndRecord, mode: .default, options: [.defaultToSpeaker])
-            try AVAudioSession.sharedInstance().setMode(.default)
+            // try AVAudioSession.sharedInstance().setCategory(.playback, mode: .default, options: [])
```
Comment (AdamRa97) [inline SmartWeights/SmartWeights/Workout/StartWorkout.swift:466]: If these are unnecessary, just remove em

File: SmartWeights/SmartWeights/Workout/WorkoutMainPage.swift [hunk 4]
```diff
@@ -266,15 +266,22 @@ struct WorkoutMainPage: View {
...
             
             // Start/Reset workout button
-            //----------------------BUTTON ACTION---------------------// 
+            //----------------------BUTTON ACTION---------------------//
             Button(action: {
                 print(".............THIS IS BUTTON TEXT",buttonText)
                 if viewModel.hasWorkoutStarted {
                     if buttonText == "Finish Workout" {
                         
+                        // CODE TO UPDATE WORKOUTS ACHIEVEMENTS (1st Workout, Workout Machine, Perfect Form)
```
Comment (AdamRa97) [inline SmartWeights/SmartWeights/Workout/WorkoutMainPage.swift:275]: I think we should move these to finishWorkout()

**Review by AdamRa97 — COMMENTED** [review]
read my comments

File: SmartWeights/SmartWeights/Workout/StartWorkout.swift (hunk 3 above)
Comment (SlowMiata) [reply SmartWeights/SmartWeights/Workout/StartWorkout.swift:466]: mb i thought i removed it already

File: SmartWeights/SmartWeights/Workout/StartWorkout.swift (hunk 2 above)
Comment (SlowMiata) [reply SmartWeights/SmartWeights/Workout/StartWorkout.swift:400]: these should be there.

**Pull request merged**